lib/Runtime/Language/JavascriptOperators.cpp

```diff
@@ -10033,6 +10032,24 @@ namespace Js
         return true;
     }
 
+    bool JavascriptOperators::CheckIfPrototypeChainContainsProxyObject(RecyclableObject* prototype)
+    {
+        Assert(JavascriptOperators::IsObjectOrNull(prototype));
+
+        if (prototype->GetTypeId() == TypeIds_Null)
+        {
+            return false;
+        }
+        else if (prototype->GetTypeId() == TypeIds_Proxy)
+        {
+            return true;
+        }
+        else
+        {
+            return CheckIfPrototypeChainContainsProxyObject(prototype->GetPrototype());
```

Comment:
Agreed.  We use iterative loops for prototype chain walking almost everywhere.  I think we have one or two recursive walks, but I would favor a loop.

---
In reply to: [61649427](https://github.com/Microsoft/ChakraCore/pull/886#discussion_r61649427) [](ancestors = 61649427)